app/delegate-server/Contract/Commit.purs

```diff
@@ -14,46 +20,124 @@ module DelegateServer.Contract.Commit
   , signTransaction
   , submit
   )
+import Contract.TxConstraints (TxConstraints)
+import Contract.TxConstraints
+  ( mustBeSignedBy
+  , mustSpendPubKeyOutput
+  , mustSpendScriptOutput
+  , mustValidateIn
+  ) as Constraints
+import Contract.UnbalancedTx (UnbalancedTx(UnbalancedTx), mkUnbalancedTx)
+import Contract.Wallet (ownPaymentPubKeyHash)
 import Control.Error.Util ((!?), (??))
-import Control.Monad.Except (ExceptT(ExceptT), withExceptT)
-import Data.Argonaut (encodeJson)
-import Data.Codec.Argonaut (JsonCodec, string) as CA
-import Data.Codec.Argonaut.Variant (variantMatch) as CAV
-import Data.Generic.Rep (class Generic)
-import Data.Newtype (unwrap)
-import Data.Profunctor (dimap)
+import Control.Monad.Except (ExceptT(ExceptT), mapExceptT, throwError, withExceptT)
+import Control.Monad.Trans.Class (lift)
+import Ctl.Internal.BalanceTx.RedeemerIndex
+  ( attachIndexedRedeemers
+  , indexRedeemers
+  , mkRedeemersContext
+  )
+import Data.Argonaut (Json, encodeJson)
+import Data.Codec.Argonaut (JsonCodec, encode) as CA
+import Data.Codec.Argonaut.Generic (nullarySum) as CAG
+import Data.Map (fromFoldable) as Map
 import Data.Show.Generic (genericShow)
-import Data.Tuple.Nested (type (/\), (/\))
-import Data.Variant (inj, match) as Variant
 import DelegateServer.App (runContract, runContractLift)
+import DelegateServer.Handlers.SignCommitTx (signCommitTxErrorCodec)
+import DelegateServer.Helpers (modifyF)
 import DelegateServer.HydraNodeApi.Http (commit)
-import DelegateServer.HydraNodeApi.Types.Commit
-  ( CommitUtxoMap
-  , mkCollateralCommit
-  , mkStandingBidCommit
-  )
 import DelegateServer.Lib.ServerConfig (mkLocalhostHttpServerConfig)
-import DelegateServer.Lib.Transaction (reSignTransaction)
+import DelegateServer.Lib.Transaction (appendTxSignatures, reSignTransaction, setAuxDataHash)
 import DelegateServer.Lib.Wallet (withWallet)
+import DelegateServer.PeerDelegate.Http (signCommitTx)
 import DelegateServer.State (class AppBase, class AppInit, access, readAppState)
+import DelegateServer.Types.HydraCommitRequest (mkFullCommitRequest, mkSimpleCommitRequest)
+import DelegateServer.Types.HydraHeadPeer (HydraHeadPeer)
+import DelegateServer.Types.ServerResponse
+  ( ServerResponse(ServerResponseSuccess, ServerResponseError)
+  )
 import HydraAuctionOffchain.Contract.QueryUtxo (queryStandingBidUtxo)
-import HydraAuctionOffchain.Contract.Types (StandingBidState)
+import HydraAuctionOffchain.Contract.Types
+  ( AuctionInfoRec
+  , AuctionTerms(AuctionTerms)
+  , StandingBidRedeemer(MoveToHydraRedeemer)
+  , StandingBidState
+  , Utxo
+  )
 import HydraAuctionOffchain.Contract.Validators (mkStandingBidValidator)
+import HydraAuctionOffchain.Helpers ((!*))
 import HydraAuctionOffchain.Lib.Json (printJson)
 import HydraAuctionOffchain.Service.Common (ServiceError)
+import JS.BigInt (fromInt) as BigInt
 import Type.Proxy (Proxy(Proxy))
 
+buildCommitTx
+  :: forall m. AppBase m => Json -> ExceptT ServiceError m BalancedSignedTransaction
```

Comment:
Why do we use `Json` here? Can a more specific type be beneficial here?

Reply:
Addressed in https://github.com/mlabs-haskell/hydra-auction-offchain/pull/41/commits/e7a7d224fcd2078802570eb8890a047f625037c6